readelf doesn't reset dynamic_info_DT_GNU_HASH

 http://sourceware.org/bugzilla/show_bug.cgi?id=12523

===================================================================
RCS file: /cvs/src/src/binutils/ChangeLog,v
retrieving revision 1.1708.2.8
retrieving revision 1.1708.2.9
diff -u -r1.1708.2.8 -r1.1708.2.9
--- src/binutils/ChangeLog	2011/02/01 12:25:36	1.1708.2.8
+++ src/binutils/ChangeLog	2011/03/01 00:02:38	1.1708.2.9
@@ -1,3 +1,8 @@
+2011-03-01  Akos Pasztory  <akos.pasztory@gmail.com>
+
+	PR binutils/12523
+	* readelf.c (process_object): Clear dynamic_info_DT_GNU_HASH.
+
 2010-08-13  Dan Rosenberg  <dan.j.rosenberg@gmail.com>
 
 	PR binutils/11889
===================================================================
RCS file: /cvs/src/src/binutils/readelf.c,v
retrieving revision 1.522.2.1
retrieving revision 1.522.2.2
diff -u -r1.522.2.1 -r1.522.2.2
--- src/binutils/readelf.c	2011/02/01 12:25:36	1.522.2.1
+++ src/binutils/readelf.c	2011/03/01 00:02:38	1.522.2.2
@@ -12287,6 +12287,7 @@
 
   for (i = ARRAY_SIZE (dynamic_info); i--;)
     dynamic_info[i] = 0;
+  dynamic_info_DT_GNU_HASH = 0;
 
   /* Process the file.  */
   if (show_name)
-- 
